builder/osc/bsuvolume/artifact_test.go

```diff
@@ -0,0 +1,94 @@
+package bsuvolume
+
+import (
+	"reflect"
+	"testing"
+
+	registryimage "github.com/hashicorp/packer-plugin-sdk/packer/registry/image"
+	"github.com/mitchellh/mapstructure"
+)
+
+func TestArtifactState(t *testing.T) {
+	expectedData := "this is the data"
+	artifact := &Artifact{
+		StateData: map[string]interface{}{"state_data": "this is the data"},
+	}
+
+	// Valid state
+	result := artifact.State("state_data")
+	if result != expectedData {
+		t.Fatalf("Bad: State data was %s instead of %s", result, expectedData)
+	}
+
+	// Invalid state
+	result = artifact.State("invalid_key")
+	if result != nil {
+		t.Fatalf("Bad: State should be nil for invalid state data name")
+	}
+
+	// Nil StateData should not fail and should return nil
+	artifact = &Artifact{}
+	result = artifact.State("key")
+	if result != nil {
+		t.Fatalf("Bad: State should be nil for nil StateData")
+	}
+}
+
+func TestArtifactState_hcpPackerRegistryMetadata(t *testing.T) {
+
+	volumes := make(BsuVolumes)
+	volumes["west"] = []string{"vol-4567", "vol-0987"}
```

Comment:
`west` is the region right ? maybe use an existing region like `eu-west-2` instead ? (but maybe regions work differently on packer ?)